src/PostgREST/App.hs

```diff
@@ -332,10 +333,10 @@ addFilter (path, flt) (Node rn forest) =
       where maybeNode = find ((name==).fst.snd.rootLabel) forst
 
 toSourceRelation :: Text -> Relation -> Maybe Relation
-toSourceRelation mt r@(Relation _ t _ ft _ _ rt _ _)
-  | mt == t = Just $ r {relTable=sourceSubqueryName}
-  | mt == ft = Just $ r {relFTable=sourceSubqueryName}
-  | Just mt == rt = Just $ r {relLTable=Just sourceSubqueryName}
+toSourceRelation mt r@(Relation t _ ft _ _ rt _ _)
```

Comment:
Wish I could help here, but when I switched all the types I just went through all the things that were throwing compilation errors and fixing them. If you think that's confusing just take a look at [this](https://github.com/calebmer/postgrest/commit/acb8f8a15491352fcad6beda7b1df6f9515822fc) fix for the failing test.